dev/src/index.ts

```diff
@@ -1038,9 +1038,12 @@ export class Firestore {
    * @private
    */
   private createCallOptions(): CallOptions {
-    const gaxHeaders: CallOptions = {otherArgs: {headers: {}}};
-    gaxHeaders.otherArgs!.headers[CLOUD_RESOURCE_HEADER] = this.formattedName;
-    return gaxHeaders;
+    const headers: any = {};
+    headers[CLOUD_RESOURCE_HEADER] = this.formattedName;
+    if (this._settings.additionalHeaders) {
+      Object.assign(headers, this._settings.additionalHeaders);
+    }
+    return {otherArgs: { headers }};
```

Comment:
This is pretty much what I had stashed away in my local client as well.

Can we make this a little more ES-6y by changing this to:

```
 return {
      otherArgs: {
        headers: {
          [CLOUD_RESOURCE_HEADER]: this.formattedName,
          ...this._settings.customHeaders
        }
      }
    };
```

This will remove the usage of `any`. Note that I also changed the  `additionalHeaders` name to `customHeaders`, which is just a slight preference.

Reply:
holy wow, you're a wizard